src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
+ * {@link Actions} for a full example of how actions are defined, and how an application can listen to and handle them.
+ *
+ * This can be for the notification button(s). Later versions of the service will add additional control types. All
+ * actions, for all control types, will be returned to the application via the same `notification-action` event type.
+ * Check the contents of the event object for details on what triggered this action, which control the user
+ * interacted with, and what metadata was attached to the action.
  *
- * This will not fire in cases of Notification Buttons being clicked.  See {@link NotificationButtonClickedEvent}.
+ * This type includes a generic type argument, should applications wish to define their own interface for action
+ * results. See {@link NotificationActionResult} for details.
  *
  * @event
  */
-export interface NotificationClickedEvent {
-    type: 'notification-clicked';
-    notification: Notification;
+export interface NotificationActionEvent<T = {}> {
+    type: 'notification-action';
+
+    /**
+     * The notification that created this action
+     */
+    notification: Readonly<Notification>;
+
+    /**
+     * Indicates what triggered this action.
+     *
+     * Note that the `programmatic` trigger is not yet implemented.
+     */
+    trigger: ActionTrigger;
+
+    /**
+     * The control whose interaction resulted in this action being raised. Will only be present when {@link trigger} is
+     * {@link ActionTrigger.CONTROL}.
+     *
+     * Future versions of the service will add additional controls beyond buttons, and interactions with these new
+     * control types will also come through this one event type. For best forward-compatibility, applications should
+     * always check the `type` property of this control, and not assume that the type will always be `'button'`.
+     *
+     * This field is marked optional as future versions of the service will also include alternate methods of raising
+     * `notification-action` events that do not originate from a button or other control.
+     *
+     * When present, the object here will always be strictly equal to one of the control definitions within
+     * `notification`. This means `indexOf` checks and other equality checks can be performed on this field if
+     * required, such as:
+     *
+     * ```ts
+     * function onNotificationAction(event: NotificationActionEvent): void {
+     *     if (event.control && event.control.type === 'button') {
+     *         const butttonIndex = event.notification.buttons.indexOf(event.control);
+     *
+     *         // Handle button click
+     *         // . . .
+     *     }
+     * }
+     * ```
+     */
+    control?: Readonly<ControlOptions>;
+
+    /**
+     * Application-defined metadata that this event is passing back to the application.
+     *
+     * A `notification-action` event is only raised if the {@link NotificationOptions|notification definition} includes
```

Comment:
Would go with "A notification-action event is only fired for an interaction with a control if the notification options included an action result for that interaction"

Reply:
Tweaked wording slightly as there are also non-control user interactions (select/close).